### PR TITLE
Finish simplifying data structures.

### PR DESCRIPTION
**Pull request by eirrgang**
* Replace MultiPair with PairDataCollection.
* Remove brer.metadata module.

We also simplify the creation/initialization methods,
normalize interfaces as much as possible to current Python patterns,
and try to conform to conventional interfaces for dataclasses and
serializable data structures.

* `RunData.create_from()`: new overloaded creation method from state file.
* `RunData.from_dictionary()`: removed (unused)
* `RunData.load_config()`: removed (unused)
* `RunData.clear_pair_data()`: removed (unused)
* `RunData.from_pair_data()`: removed (use case refactored)
* `PairParams.load_sites()`: removed (unused)
* `MultiPair.read_from_json()`: replaced with
  `PairDataCollection.create_from()`

Fixes https://github.com/kassonlab/brer-md/issues/10

### CODE REVIEW

File: tests/test_run_config.py [hunk 1]
```diff
@@ -180,5 +180,5 @@ def test_mpi_ensemble(tmpdir, pair_data_file, simulation_input):
         rc.run(threads=4, max_hours=0.001)
 
         if comm.Get_size() > 1:
-            # TODO: Confirm that we actually ran different ensemble members.
+            # TODO(#7): Confirm that we actually ran different ensemble members.
```
Comment (eirrgang) [inline tests/test_run_config.py:183]:
```suggestion
            # TODO(https://github.com/kassonlab/run_brer/issues/7): Confirm that we actually ran different ensemble members.
```

File: src/brer/run_data.py [hunk 2]
```diff
@@ -243,13 +261,70 @@ def save_config(self, fnm='state.json'):
...
             json.dump(self.as_dictionary(), fh, indent=4)
 
-    def load_config(self, fnm='state.json'):
-        """Load state parameters from file.
+    @typing.overload
+    @classmethod
+    def create_from(cls,
+                    source: typing.Union[str, os.PathLike, pathlib.Path],
+                    ensemble_num: int = None) -> 'RunData':
+        ...
```
Comment (eirrgang) [inline src/brer/run_data.py:269]: I should probably update the codecov configuration to ignore such lines.

**Comment by eirrgang** [comment]
I'm going to let this stew for a few days. I removed some methods that did not seem to be used. Though it seems unlikely, it is conceivable that some might be in use in some scripts in the wild.

I would appreciate some review to confirm that these changes don't produce any likely user-facing surprises.

**Review by carolinedav — APPROVED** [review]
assuming that `from_dictionary` and `from_pair_data` are unnecessary then the changes look good to me!

**Comment by eirrgang** [comment]
Both should now be served by the create_from creation method. I could not
find a case where it seemed important to initialize in several steps rather
than all at once.

Please review your scripts to make sure you aren't using these, but i don't
think there was a user facing use case.

**Comment by carolinedav** [comment]
Yeah! I don't use those functions anywhere in my code so I'm good to go!

**Pull request merged**